fixtures/auth_client.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+from rest_framework.test import APIClient
+from users.serializers import UserSerializer
+
+@pytest.fixture
+def auth_client(user = None):
```

Comment:
this fixture shouldn't create a user it should only return an authenticated client, the user creation should another fixture. if you want to do this create a fixture that creates a user then create another fixture that looks like this:
```python
@pytest.fixture
def get_client(user):
    def api_client(user_instance=None):
        if user_instance is None:
            client = APIClient()

            login_response = client.post('/login/', dict(username="ahmed", password="Ahmed_1234"))

            token = login_response.data["token"]

            client.credentials(HTTP_AUTHORIZATION='Token ' + token)

            return client, user.id
        else:
            client = APIClient()
            
            random_user = User.objects.create_user(user_instance['username'], user_instance['email'], user_instance['password'])

            login_response = client.post('/login/', dict(username=user_instance["username"], password=user_instance["password"]))

            token = login_response.data["token"]

            client.credentials(HTTP_AUTHORIZATION='Token ' + token)

            return client, random_user.id
    
    return api_client
```